extension/app/src/lib/conversation.ts

```diff
@@ -265,3 +277,38 @@ export async function postMessage({
 
   return new Ok(await mRes.json());
 }
+
+export async function handleRequestUserDataParams(
+  owner: LightWorkspaceType,
+  conversationId: string,
+  actionId: string,
+  requestedData: string[]
+) {
+  const token = await getAccessToken();
+  const url = `${process.env.DUST_DOMAIN}/api/v1/w/${owner.sId}/assistant/conversations/${conversationId}/actions/${actionId}/requested_data`;
+
+  if (!token) {
+    throw new Error("No access token found");
+  }
+
+  const res = await sendGetActiveTabMessage();
+
+  const responses: Record<string, string> = {
+    page_content: res.content,
+    page_url: res.url,
+  };
+
+  const outputs = requestedData.map((key) => {
+    return { name: key, value: responses[key] };
+  });
+
+  const body = JSON.stringify({ outputs });
+  await fetch(url, {
```

Comment:
(not urgent) can you move this to `DustApi` to keep all calls to the public API in the client.

note: you have a "request()" method for custom calls.